src/types.ts

```diff
@@ -240,8 +253,66 @@ export enum ANT_INTERACTION_TYPES {
   REMOVE_RECORD = 'Delete Record',
   TRANSFER = 'Transfer',
   BALANCE = 'Balance',
+  CREATE = 'Create Arweave Name Token',
 }
 
+export type ContractType = (typeof CONTRACT_TYPES)[keyof typeof CONTRACT_TYPES];
+export type AntInteraction =
+  (typeof ANT_INTERACTION_TYPES)[keyof typeof ANT_INTERACTION_TYPES];
+export type RegistryInteraction =
+  (typeof REGISTRY_INTERACTION_TYPES)[keyof typeof REGISTRY_INTERACTION_TYPES];
+
+export type TransactionData = {
+  [x: string]: any;
```

Comment:
nit: union types for various transaction data inputs

`BuyRecordPayload` vs. `TransferRecord` payload, etc. etc.

Reply:
added